src/main/java/algorithms/curated170/hard/CoinPath.java

```diff
@@ -0,0 +1,53 @@
+package algorithms.curated170.hard;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class CoinPath {
+    public List<Integer> cheapestJump(int[] A, int B) {
+        int coins = A.length;
+        if (A == null || coins < 1 || A[coins - 1] < 0) {
+            return Collections.emptyList();
+        }
+        int[] forwardPathData = new int[coins];
+        forwardPathData[coins - 1] = -1;
+
+        for (int i = coins - 2; i >= 0; i--) {
+            if (A[i] == -1) {
+                continue;
+            }
+
+            int minCost = Integer.MAX_VALUE;
+            int jumpingLimit = Math.min(i + B, coins - 1);
+
+            for (int j = i + 1; j <= jumpingLimit; j++) {
+                if (A[j] == -1) {
+                    continue;
+                }
+                if (A[j] < minCost) {
+                    minCost = A[j];
+                    forwardPathData[i] = j;
+                }
+            }
+
+            if (minCost != Integer.MAX_VALUE) {
+                A[i] += minCost;
+            } else {
+                A[i] = -1;
+            }
+        }
+
+        if (A[0] == -1) {
+            return Collections.emptyList();
```

Comment:
```suggestion
            return List.of();
```
Suggestion for the same reason.